attachments/tests.py

```diff
@@ -1,92 +1,103 @@
-        self.bob = User(username="bob")
+        self.bob = User.objects.create(username="bob")
+        self.bob.set_password('pw')
         self.bob.save()
-
-        self.tm = TestModel(name="Test1")
-        self.tm.save()
-        self.tm2 = TestModel(name="Test2")
-        self.tm2.save()
-
-        self.test_file1 = NamedTemporaryFile('w+')
-        self.test_file1.write("some test text")
-        self.test_file1.flush()
-        self.test_file1.seek(0)
-
-        self.test_file2 = NamedTemporaryFile('w+')
-        self.test_file2.write("some test text")
-        self.test_file2.flush()
-        self.test_file2.seek(0)
-
-    def testDeepCopying(self):
+        assert self.client.login(username='bob', password='pw')
+
+        self.content_type = ContentType.objects.get_for_model(TestModel)
+        self.tm = TestModel.objects.create(name="Test1")
+        self.tm2 = TestModel.objects.create(name="Test2")
+
+    @contextmanager
+    def open(self, text=None):
+        if text is None:
+            text = 'some test text'
+        with NamedTemporaryFile('w+') as f:
+            f.write(text)
+            f.flush()
+            f.seek(0)
+            yield File(f)
+
+    def create_attachment(self, obj, **kwargs):
+        att1 = Attachment.objects.create_for_object(obj, **kwargs)
+        with self.open() as f:
+            att1.file.save('models.py', f)
+        return att1
+
+    def test_deep_copying(self):
         """
         Test that doing a deep copy of a file actually attempt to create a
         second version of a file.
         """
-        att1 = Attachment.objects.create_for_object(
-            self.tm, file=self.test_file1, attached_by=self.bob,
-            title="Something", summary="Something")
-        f = File(self.test_file1)
-        att1.file.save('models.py', f)
+        att1 = self.create_attachment(
+            self.tm,
+            attached_by=self.bob,
+            title="Something",
+            summary="Something",
+        )
 
-        att2 = att1.copy(self.tm2, deepcopy=True)
+        att1.copy(self.tm2, deepcopy=True)
 
         # Ensure the saved_copy uses its proper file path
         attachments = Attachment.objects.attachments_for_object(self.tm2)
         for attachment in attachments:
             self.assertEqual(
                 attachment.file.name,
-                Attachment.get_attachment_dir(attachment,
-                                              attachment.file_name())
-            )
\ No newline at end of file
+                Attachment.get_attachment_dir(
+                    attachment,
+                    attachment.file_name(),
+                ),
+            )
+
+    def test_view_smoke_test(self):
+        url = reverse(
+            'attachment_list',
+            kwargs={
+                'content_type': self.content_type.pk,
+                'object_id': self.tm.pk,
+            },
+        )
+        r = self.client.get(url)
+        self.assertEqual(r.status_code, 200)
```

Comment:
I was only trying to add tests would result in all the files being loaded (showing us import errors). I checked all the views to make sure everything was kosher in the them.